File: leetcode/2001-3000/2401-2500/2425-bitwise-xor-of-all-pairings/BitwiseXorOfAllPairings.ts
function xorAllNums(lhs: number[], rhs: number[]): number {
  let x: number = 0;

  if (lhs.length % 2 === 1) {
    for (const r of rhs) {
      x ^= r;
    }
  }

  if (rhs.length % 2 === 1) {
    for (const l of lhs) {
      x ^= l;
    }
  }

  return x;
};

console.log(xorAllNums([2, 1, 3], [10, 2, 5, 0]))

File: leetcode/2001-3000/2401-2500/2421-2430/2425-bitwise-xor-of-all-pairings/BitwiseXorOfAllPairings.java
public class BitwiseXorOfAllPairings {
  public static void main(String[] args) {

    System.out.println(new Solution().xorAllNums(new int[] { 2, 1, 3 }, new int[] { 10, 2, 5, 0 }));
  }
}

class Solution {
  public int xorAllNums(int[] lhs, int[] rhs) {
    int left = 0;
    for (int i = 0; i < lhs.length; i++) {
      left ^= lhs[i];
    }

    int right = 0;
    for (int i = 0; i < rhs.length; i++) {
      right ^= rhs[i];
    }

    int x = 0;

    if (rhs.length % 2 == 1) {
      x ^= left;
    }

    if (lhs.length % 2 == 1) {
      x ^= right;
    }

    return x;
  }
}
